src/printing/tests.rs

```diff
@@ -0,0 +1,82 @@
+use proc_macro2::{Literal, TokenTree};
+use quote::ToTokens;
+
+use crate::{ByteLit, ByteStringLit, CharLit, FloatLit, IntegerLit, StringLit};
+
+#[test]
+fn it_preserves_bool_true() {
+    let other = crate::BoolLit::True
+        .to_token_stream()
+        .into_iter()
+        .next()
+        .and_then(|v| {
+            if let TokenTree::Ident(v) = v {
+                Some(v)
+            } else {
+                None
+            }
+        })
+        .unwrap();
+    assert_eq!(other, "true")
+}
+#[test]
+fn it_preserves_bool_false() {
+    let other = crate::BoolLit::False
+        .to_token_stream()
+        .into_iter()
+        .next()
+        .and_then(|v| {
+            if let TokenTree::Ident(v) = v {
+                Some(v)
+            } else {
+                None
+            }
+        })
+        .unwrap();
+    assert_eq!(other, "false")
+}
+
+// NOTE: sometimes the round-trip to quote simplifies literals (for example float literals)
+// this is something that has to be taken into consideration when writing tests
+macro_rules! preservation {
+    ($name:ident : $ty:ident : $($content:tt)+) => {
+        #[test]
+        fn $name() {
+            use std::convert::TryFrom;
+
+            let lit = Literal::$($content)+;
+
+            let lhs = $ty::try_from(lit).unwrap();
+
+            let rhs = $ty::try_from(
+                lhs.to_token_stream()
+                    .into_iter()
+                    .next()
+                    .and_then(|v| {
+                        if let TokenTree::Literal(v) = v {
+                            Some(v)
+                        } else {
+                            None
+                        }
+                    })
```

Comment:
That shouldn't be necessary as a `TryFrom` impl also exists for `TokenTree`, not only `Literal`, right?